src/test/java/de/tum/in/www1/artemis/PlagiarismCheckForExcercisesIntegrationTest.java

```diff
@@ -0,0 +1,93 @@
+package de.tum.in.www1.artemis;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.enumeration.ExerciseType;
+import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismComparison;
+import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismResult;
+import de.tum.in.www1.artemis.domain.plagiarism.modeling.ModelingSubmissionElement;
+import de.tum.in.www1.artemis.domain.plagiarism.text.TextSubmissionElement;
+import de.tum.in.www1.artemis.util.FileUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.util.LinkedMultiValueMap;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PlagiarismCheckForExcercisesIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    private Course course;
+
+    private int studentAmount;
+
+    @BeforeEach
+    public void initTestCase() throws IOException {
+        String submissionText = "Lorem ipsum dolor sit amet, consectetur adipiscing elit";
+        String submissionModel = FileUtils.loadFileFromResources("test-data/model-submission/model.54727.json");
+        studentAmount = 25;
+
+        course = database.addCourseWithOneFinishedTextExerciseAndSimilarSubmissions(submissionText, studentAmount);
+        database.addCourseWithOneFinishedModelingExerciseAndSimilarSubmissions(submissionModel, studentAmount, course);
+
+    }
+
+    @AfterEach
+    public void tearDown() {
+        database.resetDatabase();
+    }
+
+    @Test
+    @WithMockUser(username = "editor1", roles = "EDITOR")
+    public void testCheckPlagiarismResultForTextExercise() throws Exception {
```

Comment:
The tests are nearly identical. Can you maybe extract the common parts into helper methods to reduce duplicated code?

Reply:
Can you please also remove the public modifier to conform with https://github.com/ls1intum/Artemis/pull/5324/

Reply:
I will do this in a follow up PR

Reply:
If I change the public to private the test is not recognized, I think it mus be first cahnged to JUNit 5 or how can I test it otrherwise?

Reply:
Don't change it to private, change it to package-private (no modifier)

Reply:
done